src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -44,71 +44,96 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
+        self.package_manager_max_retries = 5
+        self.error_codes_to_retry = [self.zypper_exitcode_zypp_locked]
 
     def refresh_repo(self):
         self.composite_logger.log("Refreshing local repo...")
         # self.invoke_package_manager(self.repo_clean)  # purges local metadata for rebuild - addresses a possible customer environment error
-        for i in range(0, Constants.MAX_ZYPPER_REPO_REFRESH_RETRY_COUNT):
-            try:
-                self.invoke_package_manager(self.repo_refresh)
-                return
-            except Exception as error:
-                if i < Constants.MAX_ZYPPER_REPO_REFRESH_RETRY_COUNT - 1:
-                    self.composite_logger.log_warning("Exception on package manager refresh repo. [Exception={0}] [RetryCount={1}]".format(repr(error), str(i)))
-                    time.sleep(pow(2, i) + 1)
-                else:
-                    if Constants.ERROR_ADDED_TO_STATUS in repr(error):
-                        error.args = error.args[:1]  # remove Constants.ERROR_ADDED_TO_STATUS flag to add new message to status
+        try:
+            self.invoke_package_manager(self.repo_refresh)
+            return
+        except Exception as error:
+            if Constants.ERROR_ADDED_TO_STATUS in repr(error):
+                error.args = error.args[:1]  # remove Constants.ERROR_ADDED_TO_STATUS flag to add new message to status
 
-                    error_msg = "Unable to refresh repo (retries exhausted). [{0}] [RetryCount={1}]".format(repr(error), str(i))
+            error_msg = "Unable to refresh repo (retries exhausted). [{0}]".format(repr(error))
 
-                    # Reboot if not already done
-                    if self.status_handler.get_installation_reboot_status() == Constants.RebootStatus.COMPLETED:
-                        error_msg = "Unable to refresh repo (retries exhausted after reboot). [{0}] [RetryCount={1}]".format(repr(error), str(i))
-                    else:
-                        self.composite_logger.log_warning("Setting force_reboot flag to True.")
-                        self.force_reboot = True
-                        
-                    self.composite_logger.log_warning(error_msg)
-                    self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
+            # Reboot if not already done
+            if self.status_handler.get_installation_reboot_status() == Constants.RebootStatus.COMPLETED:
+                error_msg = "Unable to refresh repo (retries exhausted after reboot). [{0}]".format(repr(error))
+            else:
+                self.composite_logger.log_warning("Setting force_reboot flag to True.")
+                self.force_reboot = True
+                
+            self.composite_logger.log_warning(error_msg)
+            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
```

Comment:
Q: the error coming from command invocation, will be added to error objects from within package manager, either in line 110 or 132 in this file. Why do we need to add another error message?

Some context on error objects:
Error objects refers to the errors json within any summary substatus (i.e. PatchInstallationSummary or PatchAssesmentSummary, etc). Since it's added to a json file with possible size limitations, we only add contents actionable contents within it, which would help the user understand the cause of failure.

For eg: in this case, a message stating which command failed (from line 110 or 132). refresh_repo is a function defined in our code and adding a message "Unable to refresh repo (retries exhausted after reboot)" will not be of much help to the user. 

Add this error message to our logs but don't add it to the error objects, since we already have details on which command failed in there.

To do this, keep Constants.ERROR_ADDED_TO_STATUS in the error_message and simple use composite_logger (any log type you prefer), This will add the message to log files but not error objects in the status

Reply:
I added this extra error message so that it the source of error could be extra visible - in this case, the error message could say that retries were exhausted after reboot. But I understand now what you mean with the extra context, and I will put the error message from this area into a warning log and keep the Constants.ERROR_ADDED_TO_STATUS in the message.